src/Components/Web.JS/src/Services/EventRegistry.ts

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { IBlazor } from '../GlobalExports';
+
+// The base Blazor event type.
+// Properties listed here get assigned by the event registry in 'dispatchEvent'.
+interface BlazorEvent {
+  blazor: IBlazor;
+  type: keyof BlazorEventMap;
+}
```

Comment:
This is interesting. If I'm reading correctly, your use of `blazor` here is for forward compatibility with some possible future where we pass the Blazor instance to the callback?

Presumably this is to handle the multiple-Blazor case.

But if so, wouldn't that already be covered by having a different event registration call for each Blazor instance? That would be more compatible, since the different Blazor instances could be of different versions and hence have completely different APIs. Or is there a different reason for doing this?

Reply:
If it's for internal use only (and I see you used `Omit` below, perhaps with that in mind) then it would certainly be fine either way.

Reply:
I agree, if this is for an event, I don't see why we would have `blazor` as an arg, given that you know where you are registering.

The other bit (which I am a bit unsure is whether this should have a similar shape as a `https://developer.mozilla.org/en-US/docs/Web/API/Event` and have some of it's behaviors, like `this` being associated with the event in the handler, and so on.

Reply:
> This is interesting. If I'm reading correctly, your use of blazor here is for forward compatibility with some possible future where we pass the Blazor instance to the callback?

Yeah, that's correct. I added initially added this when taking the `document.addEventListener` approach (instead of the current approach, which is `Blazor.addEventListener`), so that developers could detect which `Blazor` instance emitted the event. But since the event is now getting registered on the `Blazor` instance itself, it's probably fine that we remove this.

> The other bit (which I am a bit unsure) is whether this should have a similar shape as a https://developer.mozilla.org/en-US/docs/Web/API/Event

Couldn't we extend this API in the future in a non-breaking manner to emulate the shape of DOM events a bit more? Or is this something you think we need to decide on now?

Reply:
> Couldn't we extend this API in the future in a non-breaking manner to emulate the shape of DOM events a bit more? Or is this something you think we need to decide on now?

Sure, I'd be fine with treating only the declared API as supported, and undocumented things like "the value of `this`" being open for change in the future.